ministryofjustice/laa-apply-for-criminal-legal-aid
# [CRIMAPP-441] Submit and rehydrate last jsa appointment date

## Description of change
Adds submission and rehydration for last jsa appointment attribute

## Link to relevant ticket
[CRIMAPP-441](https://dsdmoj.atlassian.net/browse/CRIMAPP-441)
[CRIMAPP-739](https://dsdmoj.atlassian.net/browse/CRIMAPP-739)

## Notes for reviewer

## Screenshots of changes (if applicable)

### Before changes:

### After changes:

<img width="1130" alt="Screenshot 2024-04-18 at 12 51 48" src="https://github.com/ministryofjustice/laa-apply-for-criminal-legal-aid/assets/51047911/0e349b0f-a4ea-475c-90f1-e41311866e2f">

## How to manually test the feature
Requires checking out of the [review](https://github.com/ministryofjustice/laa-review-criminal-legal-aid/pull/581) and [datastore](https://github.com/ministryofjustice/laa-criminal-applications-datastore/pull/243) branches to test locally

Scenario 1: JSA with last jsa appointment date
On apply:

1. Start a means tested application and fill in info until you get to the nino screen
2. Enter a nino
3. Select the JSA passporting benefit
4. Enter a last jsa appointment date
5. The dwp check should state that you do not receive a passporting benefit, select no to the result being correct, then yes to the details being correct, then select yes you can provide evidence of the passporting benefit
6. Fill in the rest of the application ensuring you have uploaded evidence (as you will be prevented from submitting)
7. Confirm the last jsa appointment date appears on the completed application page and is the correct date

In review, send back the application 
And then in apply confirm that you can see last jsa appointment date on the returned application page and is the correct date


You can also test submitting an app with another passporting benefit and an app with no passporting benefit and verifying that the attribute is nil in the rails console

[CRIMAPP-441]: https://dsdmoj.atlassian.net/browse/CRIMAPP-441?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CRIMAPP-739]: https://dsdmoj.atlassian.net/browse/CRIMAPP-739?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
